core/src/main/scala/com/spotify/featran/transformers/QuantileOutlierRejector.scala

```diff
@@ -0,0 +1,136 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import com.spotify.featran.{FeatureBuilder, FeatureRejection}
+import com.twitter.algebird._
+
+/**
+  * Transform features by passing them through, and rejecting them if they are contained in the last
+  * quantile defined by the number of buckets in the `numBuckets` parameter.
+  *
+  * The bin ranges are chosen using the Algebird's QTree approximate data structure. The precision
+  * of the approximation can be controlled with the `k` parameter.
+  *
+  * Missing values are transformed to zero vectors.
+  *
+  * Values contained in the last bucket will report [[FeatureRejection.Outlier]] as rejection.
+  *
+  * When using aggregated feature summary from a previous session, values outside of previously seen
+  * `[min, max]` will also report [[FeatureRejection.Outlier]] as rejection.
+  *
+  */
+object QuantileOutlierRejector {
+
+  /**
+    * Create a new [[QuantileOutlierRejector]] instance.
+    *
+    * @param rejectLower whether to reject outlier based on the first quantile.
+    * @param rejectUpper whether to reject outlier based on the last quantile.
+    * @param numBuckets  number of buckets (quantiles, or categories) into which data points are
+    *                    grouped, must be greater than or equal to 2
+    * @param k           precision of the underlying Algebird QTree approximation
+    */
+  def apply(name: String,
+            rejectLower: Boolean = true,
+            rejectUpper: Boolean = true,
+            numBuckets: Int = 3,
+            k: Int = QTreeAggregator.DefaultK)
+  : Transformer[Double, BaseQuantileRejector.B, BaseQuantileRejector.C] =
+    new QuantileOutlierRejector(name, rejectLower, rejectUpper, numBuckets, k)
+}
+
+private class QuantileOutlierRejector(name: String,
+                                      val rejectLower: Boolean,
+                                      val rejectUpper: Boolean,
+                                      override val numBuckets: Int,
+                                      override val k: Int)
+  extends BaseQuantileRejector(name, numBuckets, k) {
+  require(rejectLower || rejectUpper, "at least one of [rejectLower, rejectLower] must be set")
+
+  import BaseQuantileRejector.C
+
+  protected def calculateBounds(fq: Double, lq: Double): (Double, Double) = (fq, lq)
+
+  // scalastyle:off cyclomatic.complexity
+  override def buildFeatures(a: Option[Double], c: C, fb: FeatureBuilder[_]): Unit =
+    a match {
+      case Some(x) =>
+        val (fq, lq, min, max) = c
+        val (l, u) = calculateBounds(fq, lq)
+        fb.add(name, x)
+        // all elements can't be the same
+        if (x < min || x > max){
+          fb.reject(this, FeatureRejection.Outlier(x))
```

Comment:
decided to reject as `Outlier` if they are outside the seen boundary but could easily change it to `OutOfBound`